apps/bc/contracts/BC.sol

```diff
@@ -0,0 +1,353 @@
+/*
+ * SPDX-License-Identitifer:    GPL-3.0-or-later
+ */
+
+pragma solidity ^0.4.24;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/common/EtherTokenConstant.sol";
+import "@aragon/os/contracts/common/IsContract.sol";
+import "@aragon/os/contracts/common/SafeERC20.sol";
+import "@aragon/os/contracts/lib/token/ERC20.sol";
+import "@aragon/os/contracts/lib/math/SafeMath.sol";
+
+import "@aragon/apps-token-manager/contracts/TokenManager.sol";
+import "@aragon/apps-vault/contracts/Vault.sol";
+import "@aragonblack/fundraising-pool/contracts/Pool.sol";
+
+import "./BancorContracts/converter/BancorFormula.sol";
+
+import "./IBC.sol";
+
+contract BC is EtherTokenConstant, IsContract, AragonApp, IBC, BancorFormula {
+    using SafeERC20 for ERC20;
+    using SafeMath for uint256;
+
+    uint32 public ppm = 1000000;
+    uint256 public batchBlocks;
+
+    // bytes32 public constant _ROLE = keccak256("_ROLE");
```

Comment:
This is where we should define roles. The way I see it we need at least the following roles:

`BUY_ROLE` even though it will be granted to `ANY_ADDRESS`
`SELL_ROLE` even though it will be granted to `ANY_ADDRESS`
`UPDATE_CURVE_ROLE` which should be granted to the voting app [or the API contract]